src/function/statistics/quantileSeq.js

```diff
@@ -41,109 +41,112 @@ export const createQuantileSeq = /* #__PURE__ */ factory(name, dependencies, ({
    * @param {Boolean} sorted=false              is data sorted in ascending order
    * @return {Number, BigNumber, Unit, Array}   Quantile(s)
    */
-  function quantileSeq (data, probOrN, sorted) {
-    let probArr, dataArr, one
 
-    if (arguments.length < 2 || arguments.length > 3) {
-      throw new SyntaxError('Function quantileSeq requires two or three parameters')
-    }
+  return typed(name, {
+    'Array | Matrix, number | BigNumber | Unit': (data, p) => _quantileSeqProbNumber(data, p, false),
+    'Array | Matrix, number | BigNumber | Unit, boolean': _quantileSeqProbNumber,
+    'Array | Matrix, Array | Matrix': (data, p) => _quantileSeqProbCollection(data, p, false),
+    'Array | Matrix, Array | Matrix, boolean': _quantileSeqProbCollection
+  })
 
-    if (isCollection(data)) {
-      sorted = sorted || false
-      if (typeof sorted === 'boolean') {
-        dataArr = data.valueOf()
-        if (isNumber(probOrN)) {
-          if (probOrN < 0) {
-            throw new Error('N/prob must be non-negative')
-          }
+  function _quantileSeqProbNumber (data, probOrN, sorted) {
+    let probArr, one
+    const dataArr = data.valueOf()
+    if (isNumber(probOrN)) {
+      if (probOrN < 0) {
+        throw new Error('N/prob must be non-negative')
+      }
 
-          if (probOrN <= 1) {
-            // quantileSeq([a, b, c, d, ...], prob[,sorted])
-            return _quantileSeq(dataArr, probOrN, sorted)
-          }
+      if (probOrN <= 1) {
+        // quantileSeq([a, b, c, d, ...], prob[,sorted])
+        return _quantileSeq(dataArr, probOrN, sorted)
+      }
 
-          if (probOrN > 1) {
-            // quantileSeq([a, b, c, d, ...], N[,sorted])
-            if (!isInteger(probOrN)) {
-              throw new Error('N must be a positive integer')
-            }
-
-            const nPlusOne = probOrN + 1
-            probArr = new Array(probOrN)
-            for (let i = 0; i < probOrN;) {
-              probArr[i] = _quantileSeq(dataArr, (++i) / nPlusOne, sorted)
-            }
-            return probArr
-          }
+      if (probOrN > 1) {
+        // quantileSeq([a, b, c, d, ...], N[,sorted])
+        if (!isInteger(probOrN)) {
+          throw new Error('N must be a positive integer')
         }
 
-        if (isBigNumber(probOrN)) {
-          const BigNumber = probOrN.constructor
+        const nPlusOne = probOrN + 1
+        probArr = new Array(probOrN)
+        for (let i = 0; i < probOrN; i++) {
+          probArr[i] = _quantileSeq(dataArr, (i + 1) / nPlusOne, sorted)
+        }
+        return probArr
+      }
+    }
 
-          if (probOrN.isNegative()) {
-            throw new Error('N/prob must be non-negative')
-          }
+    if (isBigNumber(probOrN)) {
```

Comment:
I think a nice extra step would be to move the BigNumber implementation in a separate function, like `_quantileSeqProbBigNumber`, what do you think?

Even nicer would be to merge the two implementations into a single, generic one that can handle both `number` and `BigNumber` by replacing operations like `probOrN <= 1` with `smallerEq(probOrN, 1)`, but that is quite some extra work.

Reply:
No problem, I was wondering the same thing, I will check this and the conflict due to the other merge.

Reply:
@dvd101x do you still want to have a look into extracting the BigNumber logic of quantileSeq in a separate function?

Reply:
Hi Jos, yes. Unfortunately I haven’t found the solution yet but will have more time to work on it next week. I will report back.